File: p010.java
/* 
 * Solution to Project Euler problem 10
 * By Nayuki Minase
 * 
 * http://nayuki.eigenstate.org/page/project-euler-solutions
 * https://github.com/nayuki/Project-Euler-solutions
 */


public final class p010 implements EulerSolution {
	
	public static void main(String[] args) {
		System.out.println(new p010().run());
	}
	
	
	private static final int LIMIT = 2000000;
	
	public String run() {
		long sum = 0;
		for (int p : Library.listPrimes(LIMIT - 1))
			sum += p;
		return Long.toString(sum);
	}
	
}

File: p010.py
# 
# Solution to Project Euler problem 10
# by Project Nayuki
# 
# http://www.nayuki.io/page/project-euler-solutions
# https://github.com/nayuki/Project-Euler-solutions
# 


def compute():
	# Sieve of Eratosthenes
	ans = 0
	isprime = [True] * 2000001
	for i in range(2, len(isprime)):
		if isprime[i]:
			ans += i
			for j in range(i * i, len(isprime), i):
				isprime[j] = False
	return str(ans)


if __name__ == "__main__":
	print(compute())
